main.go

```diff
@@ -121,6 +123,9 @@ func InitFlags(fs *pflag.FlagSet) {
 	fs.DurationVar(&leaderElectionRetryPeriod, "leader-elect-retry-period", 2*time.Second,
 		"Duration the LeaderElector clients should wait between tries of actions (duration string)")
 
+	fs.StringVar(&leaderElectionResourceLock, "leader-election-resource-lock", resourcelock.ConfigMapsLeasesResourceLock, "Which resource type to use for leader election. "+
```

Comment:
Q1: should we do this in KCP/CABPK/CAPD as well?
Q2: should we reccomand providers to do the same adding this to v1alpha4 --> v1beta1 upgrade doc?

Reply:
Also see AWS analogous PR https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2822.
Yeh if we go ahead with it I agree we should address both points. Wanted to hear everyones feelings about proceeding with it first.

Reply:
I don't have strong opinions given that we ensure a reasonable default so the users can tune this parameter only if required. Also, given that clusterctl is not supporting setting this flag (but the operator most probably will), I don't see this as something we should get in v1.0

Reply:
Why should this be exposed, rather than using a sane default and locking down RBAC?

Reply:
(Couldn't we switch to leases?)

Reply:
>Why should this be exposed, rather than using a sane default and locking down RBAC?

That'd be my preference, we can do that and default to leases as far as we think the risk is understood and well documented for existing clusters https://github.com/kubernetes-sigs/controller-runtime/blob/0a9a7778cdb6d4cca3a3b9f268d90a85aa6f5fdf/pkg/leaderelection/leader_election.go#L60-L62

Reply:
For Cluster API, and all of its providers upgraded through clusterctl this should be ok given that we stop the old controllers

Reply:
(Although I do think we should do this after the release)

Reply:
sgtm